Add export feature for workspaces (#26820)
DataDog/documentation: content/en/logs/workspaces/export.md

  text: "Learn more about Log Workspaces"
- link: "/dashboards/widgets/list/"
  tag: "Documentation"
  text: "List Widget"
---

## Overview

Export your Workspaces query to dashboards and notebooks. The export feature allows you to extract datasets and display them in dashboards and notebooks for analysis or reporting. 

## Export a Workspaces dataset

From your workspace:

1. Find the dataset you want to export.  
2. Click **Save to Dashboard** or select **Save to Notebook** from the dropdown menu.  
3. In the modal, choose an existing dashboard or notebook, or add the dataset to a new dashboard or notebook.  
4. Click **Save**.  
5. (Optional) Rename the source name and click **Update**. The source name defaults to `<Workspace name> - <Cell name>`.
6. A purple banner indicates a cell that is exported.

{{< img src="logs/workspace/export/example_exported_dataset.png" alt="Example Workspaces cell that has already been exported, showing a purple banner" style="width:100%;" >}}

Saving a cell to a dashboard creates a real-time sync between the cell and every widget it's saved on. Any changes made to an exported dataset are also reflected in the associated dashboards or notebooks. 

> From the dashboard or notebook, you have the option of adjusting the columns and customizing widget options. However, you can only change dataset configurations from Log Workspaces.

Reviewer: ...the columns --> ...the displayed columns and their order [...]. However, you can only change the dataset definition (? I think that's more clear than configuration) from Log Workspaces.

Author: ```suggestion
From the dashboard or notebook, you have the option of adjusting the displayed columns and their order, and customizing widget options. However, you can only change dataset definition from Log Workspaces.
```

> From the dashboard or notebook, you have the option of adjusting the columns and customizing widget options. However, You can only change dataset configurations from Log Workspaces. 

Reviewer 2: ```suggestion
From the dashboard or notebook, you have the option of adjusting the columns and customizing widget options. However, you can only change dataset configurations from Log Workspaces.
```

> ## Further reading

Reviewer: It might be beneficial to show the widget editor side of this as well, highlighting the Edit in Log Workspaces button?

---
title: Export Log Workspaces Datasets
private: true
further_reading:
- link: "/logs/workspaces/"
  tag: "Documentation"
  text: "Learn more about Log Workspaces"
- link: "/dashboards/widgets/list/"
  tag: "Documentation"
  text: "List Widget"
---

## Overview

Export your Workspaces query to dashboards and notebooks. The export feature allows you to extract datasets and display them in dashboards and notebooks for analysis or reporting. 

> ## Export a Workspaces dataset

Reviewer 3: I think we need to align on some terminology here, in my head it's `workspace` -> `cell` -> `dataset` -> `dashboard/notebook`. 

We export a workspace `cell` as a `dataset` which can be used in notebooks/dashboards. 

Similarly, the use of `query` above this, would actually be a workspace `cell` instead.

Reviewer: Agreed, thanks for catching.

This should then read:

```
Export your Workspaces cell as a dataset to dashboards and notebooks. The export feature allows you to create datasets and display them in dashboards and notebooks for analysis or reporting. 

## Export a Workspaces cell as a dataset 
```

> Export your Workspaces query to dashboards and notebooks. The export feature allows you to extract datasets and display them in dashboards and notebooks for analysis or reporting. 
> 
> ## Export a Workspaces dataset

Author: ```suggestion
Export your Workspaces cell as a dataset to dashboards and notebooks. The export feature allows you to create datasets and display them in dashboards and notebooks for analysis or reporting. 

## Export a Workspaces cell as a dataset 
```

From your workspace:

> 1. Find the dataset you want to export.  

Reviewer: 1. Find the cell you want to export as a dataset.

Author: ```suggestion
1. Find the cell you want to export as a dataset.  
```

2. Click **Save to Dashboard** or select **Save to Notebook** from the dropdown menu.  

> 3. In the modal, choose an existing dashboard or notebook, or add the dataset to a new dashboard or notebook.  

Reviewer: add the cell as a dataset

Author: ```suggestion
3. In the modal, choose an existing dashboard or notebook, or add the cell as a dataset to a new dashboard or notebook.  
```
